### PR TITLE
Benchmarks: Add Feature - Add percentile metrics for ort and pytorch inference benchmarks

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Add 50th, 90th, 95th, 99th, 99.9th latency metrics for ORT and pytorch inference benchmarks.

### CODE REVIEW

File: superbench/benchmarks/model_benchmarks/model_base.py [hunk 1]
```diff
@@ -377,17 +377,21 @@ def __process_model_result(self, model_action, precision, step_times):
         if precision.value in precision_metric.keys():
             precision = precision_metric[precision.value]
         metric = '{}_{}_step_time'.format(precision, model_action)
+        reduce_type = ReduceType.MAX if model_action is ModelAction.TRAIN else None
         self._result.add_raw_data(metric, step_times)
-        avg = statistics.mean(step_times)
-        self._result.add_result(metric, avg, reduce_type=ReduceType.MAX if model_action is ModelAction.TRAIN else None)
+        self._result.add_result(metric, statistics.mean(step_times), reduce_type=reduce_type)
+        if model_action == ModelAction.INFERENCE:
+            self._process_percentile_result(metric, step_times, reduce_type=reduce_type)
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/model_base.py:384]: Shall we rewrite in this way?
```
if model_action == ModelAction.TRAIN:
    self._process_percentile_result(metric, step_times, reduce_type=ReduceType.MAX)
elif model_action == ModelAction.INFERENCE:
    self._process_percentile_result(metric, step_times, reduce_type=None)
else
    # report error
```

File: superbench/benchmarks/model_benchmarks/model_base.py [hunk 2]
```diff
@@ -377,17 +377,21 @@ def __process_model_result(self, model_action, precision, step_times):
...
         millisecond_per_second = 1000
         throughput = [millisecond_per_second / step_time * self._args.batch_size for step_time in step_times]
         metric = '{}_{}_throughput'.format(precision, model_action)
+        reduce_type = ReduceType.MIN if model_action is ModelAction.TRAIN else None
         self._result.add_raw_data(metric, throughput)
-        avg = statistics.mean(throughput)
-        self._result.add_result(metric, avg, reduce_type=ReduceType.MIN if model_action is ModelAction.TRAIN else None)
+        self._result.add_result(metric, statistics.mean(throughput), reduce_type=reduce_type)
+        if model_action == ModelAction.INFERENCE:
+            self._process_percentile_result(metric, throughput, reduce_type=reduce_type)
```
Comment (cp5555) [inline superbench/benchmarks/model_benchmarks/model_base.py:394]: The similar as before.

File: superbench/benchmarks/base.py [hunk 3]
```diff
@@ -8,6 +8,8 @@
 from datetime import datetime
 from abc import ABC, abstractmethod
 
+import numpy as np
```
Comment (abuccts) [inline superbench/benchmarks/base.py:11]: add it in dependency? it's only implicitly depended by torch

**Review by abuccts — APPROVED** [review]
update metrics in docs accordingly?

File: superbench/benchmarks/model_benchmarks/model_base.py (hunk 1 above)
Comment (guoshzhao) [reply superbench/benchmarks/model_benchmarks/model_base.py:384]: Do we need these metrics for training?

File: superbench/benchmarks/base.py (hunk 3 above)
Comment (guoshzhao) [reply superbench/benchmarks/base.py:11]: Added

File: superbench/benchmarks/model_benchmarks/model_base.py (hunk 1 above)
Comment (cp5555) [reply superbench/benchmarks/model_benchmarks/model_base.py:384]: Got it. My fault.

**Review by cp5555 — APPROVED** [review]

**Pull request merged**